src/Arcus.Messaging.Pumps.ServiceBus/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,175 @@
+        /// <param name="configureMessagePump">Capability to configure how the message pump should behave</param>
+        /// <returns>Collection of services to use in the application</returns>
+        public static IServiceCollection AddServiceBusTopicMessagePump<TMessagePump>(this IServiceCollection services, string topicName, string subscriptionName, Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
+            where TMessagePump : class, IHostedService
+        {
+            Guard.NotNull(services, nameof(services));
+
+            AddServiceBusMessagePump<TMessagePump>(services, topicName, subscriptionName, getConnectionStringFromSecretFunc: getConnectionStringFromSecretFunc, configureMessagePump: configureMessagePump);
+
+            return services;
+        }
+
+        /// <summary>
+        ///     Adds a message handler to consume messages from Azure Service Bus Topic
+        /// </summary>
+        /// <param name="topicName">Name of the topic to work with</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="services">Collection of services to use in the application</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="configureMessagePump">Capability to configure how the message pump should behave</param>
+        /// <returns>Collection of services to use in the application</returns>
+        public static IServiceCollection AddServiceBusTopicMessagePump<TMessagePump>(this IServiceCollection services, string topicName, string subscriptionName, Func<IConfiguration, string> getConnectionStringFromConfigurationFunc, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
+            where TMessagePump : class, IHostedService
+        {
+            Guard.NotNull(services, nameof(services));
+
+            AddServiceBusMessagePump<TMessagePump>(services, topicName, subscriptionName, getConnectionStringFromConfigurationFunc: getConnectionStringFromConfigurationFunc, configureMessagePump: configureMessagePump);
+
+            return services;
+        }
+
+        private static void AddServiceBusMessagePump<TMessagePump>(IServiceCollection services, string entityName = null, string subscriptionName = null, Func<IConfiguration, string> getConnectionStringFromConfigurationFunc = null, Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc = null, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
+            where TMessagePump : class, IHostedService
+        {
+            Guard.NotNull(services, nameof(services));
+
+            var messagePumpOptions = AzureServiceBusMessagePumpOptions.Default;
```

Comment:
Somewhere further up the code, the `AzureServiceBusMessagePump` check if we have any options configured, and takes actions accordingly; but here we always use the same default options; which makes me wonder if we would get any scenario where the message pump _will_ use the defauts ServiceBus settings.

Reply:
> Somewhere further up the code, the `AzureServiceBusMessagePump` check if we have any options configured, and takes actions accordingly; but here we always use the same default options;

Where are you referring to? Here we use the default options, and if they have configured a delegate to re-configure it it will be called so it's respected.

Reply:
The `AzureServiceBusMessagePumpSettings` has a `.Options` property, set in the ctor, that can be `null`. This will never be the case, since in the `IServiceCollectionExtensions`, we set the `.Options` to `AzureServiceBusMessagePumpOptions.Default`.

Further up the code, in the `AzureServiceBusMessagePump`, we check the `MessageHandler.Options` for `null` and assign defaults if that's the case; but, you see; that will never be the case.

Reply:
I think there are 2 things here:
1. In `IServiceCollectionExtensions` we allow clients to customize what we've defined as the default `AzureServiceBusMessagePumpSettings`
2. In `AzureServiceBusMessagePump` we use whatever was configured, if for some reason this is null, we will use `MessageHandlerOptions` their defaults which are managed by Microsoft.

Does this clarify it?

What I did do was make the properties of `AzureServiceBusMessagePumpSettings` readonly as that was not the case apparently.

Reply:
Ok, so it's actually a fail-safe or something like. Is OK! Good thinking!

Reply:
Yes indeed, sorry for the confusion!